src/core/execute.c

```diff
@@ -2446,8 +2447,12 @@ static int setup_keyring(
 
         /* And now, make the keyring owned by the service's user */
         if (uid_is_valid(uid) || gid_is_valid(gid))
-                if (keyctl(KEYCTL_CHOWN, keyring, uid, gid, 0) < 0)
-                        return log_unit_error_errno(u, errno, "Failed to change ownership of session keyring: %m");
+                if (keyctl(KEYCTL_CHOWN, keyring, uid, gid, 0) < 0) {
+                        log_error_errno(errno, "Failed to change ownership of session keyring: %m");
+                        /* well, the kernel didn't let us - clean up the invocation_id just in case */
+                        if (keyctl(KEYCTL_UNLINK, key, keyring, 0, 0) < 0)
+                                log_debug_errno(errno, "Failed to unlink (clean-up) key, after failing to change ownership: %m");
```

Comment:
hmm, i'd feel more comfortable if we'd make this fatal. if our clean-up doesn't work, then we shouldn't proceed i think. And if this is fatal this should work for you too i figure?

Reply:
yes.